src/main/resources/mappers/clickhouse/treatment/ClickhouseTreatmentMapper.xml

```diff
@@ -66,49 +66,74 @@
             )
         </where>
     </select>
-
+    
     <resultMap id="SampleTreatmentCount" type="org.cbioportal.legacy.model.SampleTreatment">
         <constructor>
             <arg column="treatment" javaType="String"/>
             <arg column="preSampleCount" javaType="_int"/>
             <arg column="postSampleCount" javaType="_int"/>
+            <arg column="preSamples" javaType="java.util.List" typeHandler="org.cbioportal.legacy.persistence.mybatis.typehandler.ClinicalEventSampleTypeHandler" />
+            <arg column="postSamples" javaType="java.util.List" typeHandler="org.cbioportal.legacy.persistence.mybatis.typehandler.ClinicalEventSampleTypeHandler" />
         </constructor>
     </resultMap>
-    <!-- Count (Pre/Post) Sample Acquisition Events for every Treatment per Patient -->
-    <select id="getSampleTreatmentCounts" resultMap="SampleTreatmentCount">
-        <!-- Nested sub query to grab minimum sample acquisition event  and also filter out duplicates -->
-        WITH sample_acquisition_events AS (
+    
+    <sql id="sampleAcquisitionEvents">
         SELECT
-        ced.value AS sample_id,
-        ced.patient_unique_id,
-        min(ced.start_date) AS time_taken,
-        ced.cancer_study_identifier AS cancer_study_identifier
+            ced.value AS sample_id,
+            ced.patient_unique_id,
+            min(ced.start_date) AS time_taken,
+            ced.cancer_study_identifier AS cancer_study_identifier
         FROM clinical_event_derived ced
         <where>
             key = 'SAMPLE_ID'
             AND (event_type ILIKE 'Sample Acquisition' OR event_type ILIKE 'SPECIMEN')
         </where>
         GROUP BY patient_unique_id, ced.value, cancer_study_identifier
-        ),
-        <!-- Nested sub query to grab minimum treatment event. When there are multiple treatments that are of the same
-        type we only care about the first treatment.
-         -->
-        treatments AS (
+    </sql>
+    
+    <sql id="treatments">
```

Comment:
doesn't this just select all treatment rows?

Reply:
I think so. This was an existing sql, I just moved this (and also `sample_acquisition_events`) into a separate SQL to make `getSampleTreatmentCounts` easier to read.